packages/http/src/mocker/__tests__/HttpMocker.spec.ts

```diff
@@ -202,7 +196,7 @@ describe('HttpMocker', () => {
             expect(response.body).toBeDefined();
 
             const allExamples = flatMap(mockResource.responses, res =>
-              flatMap(res.contents, content => content.examples),
+              flatMap(res.contents, content => content.examples || []),
```

Comment:
I'm looking at the `mockResource.responses` and I can't see a content with an undefined `examples` array. Does this `|| []` stem from a type change? Can `examples` be undefined?

If it can I'm curious why. My personal opinion about types is that if something is an array we should not let it be undefined. If there is no value it should be an empty value in my opinion. In the end what what we want to say is: there is no examples. Allowing it to be `undefined` is like saying: there can be no examples OR there can be no such thing as examples (which is confusing most of the time and forces stuff like `|| []`)

Reply:
Do not count this specific example, it's a leftover I have to fix — since there's a way to tell TypeScript "hey this specific thing is *NOT* undefined.

As your personal opinion, I am inclined to be on your side — however everybody here has one — and mine is even different. null/undefined should never appear in your code **no matter the cost** 

No:
```ts
const str: string | undefined = getTheStringThatCouldBeUndefined();
```

Yes:
```ts
const str: Option<string> = getTheStringThatCouldBeUndefined();
```

Reply:
Gotcha, I get the argument about taking up space in DB / memory though. I can live with it ;p

I'm with you on that topic btw ;)